File: src/main/java/com/github/kindrat/cassandra/client/ui/fx/component/LabeledComponentColumn.java
package com.github.kindrat.cassandra.client.ui.fx.component;

import javafx.geometry.Pos;
import javafx.scene.Node;
import javafx.scene.control.Label;
import javafx.scene.layout.HBox;
import javafx.scene.layout.VBox;

public class LabeledComponentColumn extends VBox {
    private final double spacing;

    public LabeledComponentColumn(double spacing) {
        this.spacing = spacing;
        setSpacing(spacing);
    }

    public void addLabeledElement(String label, Node element) {
        HBox row = new HBox(spacing);

        VBox labelColumn = new VBox(spacing);
        labelColumn.setAlignment(Pos.CENTER_LEFT);
        labelColumn.prefWidthProperty().bind(widthProperty().multiply(0.6));
//        UIUtil.setWidth(labelColumn, width * 0.6);
        labelColumn.getChildren().add(new Label(label));

        VBox elementColumn = new VBox(spacing);
        elementColumn.setAlignment(Pos.CENTER_LEFT);
        elementColumn.prefWidthProperty().bind(widthProperty().multiply(0.25));
//        UIUtil.setWidth(elementColumn, width * 0.25);
        elementColumn.getChildren().add(element);

        row.getChildren().addAll(labelColumn, elementColumn);
        getChildren().add(row);
    }
}

File: src/main/kotlin/com/github/kindrat/cassandra/client/ui/fx/component/LabeledComponentColumn.kt
package com.github.kindrat.cassandra.client.ui.fx.component

import javafx.geometry.Pos
import javafx.scene.Node
import javafx.scene.control.Label
import javafx.scene.layout.HBox
import javafx.scene.layout.VBox

class LabeledComponentColumn(spacing: Double) : VBox(spacing) {

    fun addLabeledElement(label: String, element: Node) {
        val row = HBox(spacing)
        val labelColumn = VBox(spacing)
        labelColumn.alignment = Pos.CENTER_LEFT
        labelColumn.prefWidthProperty().bind(widthProperty().multiply(0.6))
        labelColumn.children.add(Label(label))
        val elementColumn = VBox(spacing)
        elementColumn.alignment = Pos.CENTER_LEFT
        elementColumn.prefWidthProperty().bind(widthProperty().multiply(0.25))
        elementColumn.children.add(element)
        row.children.addAll(labelColumn, elementColumn)
        children.add(row)
    }
}